### PR TITLE
chore(data-grid): change isCell function into a type guard and remove unnecessary type assertions

### PR DESCRIPTION
**Pull request by abereghici**
change data grid `isCell` function into a type guard and remove unnecessary type assertions.

### CODE REVIEW

File: .changeset/silver-bottles-pretend.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@twilio-paste/data-grid": patch
+---
```
Comment (TheSisb) [inline .changeset/silver-bottles-pretend.md:3]: Let's patch the core bundle too

```suggestion
"@twilio-paste/data-grid": patch
"@twilio-paste/core": patch
---
```

File: packages/paste-core/components/data-grid/src/utils/cell-management.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const delayedSetFocusable = (element?: HTMLElement): void => {
  *
  * @returns {boolean}.
  */
-export const isCell = (element: Element): boolean => {
+export const isCell = (element: Element): element is HTMLTableCellElement => {
```
Comment (TheSisb) [inline packages/paste-core/components/data-grid/src/utils/cell-management.ts:41]: 💯

**Review by TheSisb — COMMENTED** [review]
Thanks for the improvement! Really nice work!

Small fix for our changeset and I can merge this today

**Comment by abereghici** [comment]
> Thanks for the improvement! Really nice work!
> 
> Small fix for our changeset and I can merge this today

done!

**Pull request merged**